vsix/CppWinRTPackageTemplateWizard.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Evaluation;
+using Microsoft.VisualStudio.TemplateWizard;
+using System.Windows.Forms;
+
+using DTEProject = EnvDTE.Project;
+using DTEProjectItem = EnvDTE.ProjectItem;
+
+namespace Microsoft.Windows.CppWinRT
+{
+    internal sealed class CppWinRTPackageTemplateWizard : IWizard
+    {
+        private IWizard wizardImpl = null;
+
+        private void ShowExceptionDialog(Exception ex)
+        {
+            string text = $"{ex.GetType().FullName}: {ex.Message}\r\n\r\n{ex.StackTrace}";
+            MessageBox.Show(text, "Caught Exception", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            throw new WizardBackoutException();
+        }
+
+        public void BeforeOpeningFile(DTEProjectItem projectItem)
+        {
+            try
+            {
+                wizardImpl?.BeforeOpeningFile(projectItem);
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void ProjectFinishedGenerating(DTEProject project)
+        {
+            try
+            {
+                VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
+
+                Project msbuildProject = new Project(project.FullName);
+                if (msbuildProject.GetPropertyValue("CppWinRTDisableAutoNuGetReference").Equals("true", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Don't forward this call to the NuGet wizard, as it has been explicitly disabled.
+                }
+                else
+                {
+                    wizardImpl?.ProjectFinishedGenerating(project);
+                }
+
+                ProjectCollection.GlobalProjectCollection.UnloadProject(msbuildProject);
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void ProjectItemFinishedGenerating(DTEProjectItem projectItem)
+        {
+            VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
And here you're not invoking the method in a try-catch block, so let's determine which way is right (: